Engine/src/Engine/InputSystem.cpp

```diff
@@ -0,0 +1,110 @@
+#include "Engine/InputSystem.h"
+#include "GLFW/glfw3.h"
+
+using namespace Engine;
+
+InputSystem* InputSystem::m_instance = nullptr;
+std::mutex InputSystem::m_instanceMutex;
+
+
+InputSystem* InputSystem::getInstance()
+{
+	std::unique_lock<std::mutex> lock;

```

Comment:
Is there a reason unique_lock is used instead of lock_guard?

Reply:
Changed it to use lock_guard